File: public/PublicMedia/static/solutions/java/Problem2.java
/*
Purpose: By considering the terms in the Fibonacci sequence whose values do not exceed four million, 
find the sum of the even-valued terms.

Author: Eben Quenneville
Date: 2022-09-03, modified 2023-01-27
*/
public class Problem2 {
    public static void main(String[] args) {
        int solution = problem2();
        assert solution == 4613732;
        System.out.println("Output: " + solution);
    }

    /**
     * @return the sum of the even-valued Fibonacci numbers < 4,000,000
     */
    public static int problem2() {
        int sum = 2;
        int first = 1;
        int previous = 2;
        int current = first + previous;
        while (current + previous < 4000000) {
            first = previous;
            previous = current;
            current = first + previous;
            if (current % 2 == 0) {
                sum += current;
            }
        }
        return sum;
    }
}

File: public/PublicMedia/static/solutions/julia/problem2.jl
# Purpose: By considering the terms in the Fibonacci sequence whose values do not exceed four million, find the sum of the even-valued terms.
#
# Author: Eben Quenneville
# Date: 2022-11-12, modified 2023-01-28

function problem2()
    sum = 2
    first = 1
    previous = 2
    current = first + previous
    while current + previous < 4000000
        first = previous
        previous = current
        current = first + previous
        if current % 2 == 0
            sum += current
        end
    end
    return sum
end

function main()
    sum = problem2()
    @assert sum == 4613732
    println("Output: ", sum)
end

main()
